Python/fileutils.c

```diff
@@ -2108,7 +2130,7 @@ static int
 join_relfile(wchar_t *buffer, size_t bufsize,
              const wchar_t *dirname, const wchar_t *relfile)
 {
-#ifdef MS_WINDOWS
+#if defined(MS_WINDOWS) && !defined(MS_WINDOWS_GAMES)
```

Comment:
What if `dirname` is "G:\\eggs" and `relfile` is the relative rooted path "\\spam"? This should be joined as "G:\\spam", not as "G:\\eggs\\spam". `relfile` could also be a relative drive path such as "T:spam", which should join as "T:spam" because it's a different drive, not as "G:\\eggs\\T:spam". Code that's meant for POSIX doesn't handle such cases. Maybe that's good enough.

Reply:
I mostly selected this implementation for it's simplicity. On the xbox you should probably simply work with absolute paths, so I considered it as good enough for now. That said a better implementation would certainly not hurt.

Reply:
On all platforms you should work with absolute paths :)

You're likely in control of your development process well enough to ensure it, but we need to account for broader expectations than that when it comes into core.

Reply:
Hm I guess on Windows/Linux we do use paths relative to the binary we ship. On Xbox basically every application has their "own" filesystem, so you really use absolute paths everywhere. I guess that might be the reason why they do not provide the API there.

Anyway I know to little about WIndows file path concatenation to come up with a better implementation.

Reply:
The POSIX code is okay as long as it's always a simple relative path such as "eggs\\foo.py". One minor annoyance is if `dirname` ends in a forward slash, such as "T:/spam/". It will add `SEP` to join it as "T:/spam/\\eggs\\foo.py". Such a path still works, so it's not a technical problem.

Reply:
I added an implementation which should be able to handle more of these cases.